PyBNN Benchmark

PhMueller opened the pull request:
First version of the pybnn benchmark. 

Benchmark is from https://github.com/automl/HPOlib1.5/blob/container/hpolib/benchmarks/ml/bnn_benchmark.py

Container is already pushed.

PhMueller committed:
[skip travis] Initial commit for the pyBNN

PhMueller committed:
[skip travis] PyBNN: create DataManager + make Benchmark deterministic

PhMueller committed:
[skip travis] PyBNN singularity recipe

PhMueller committed:
PyBNN tests

KEggensperger commented on hpolib/benchmarks/ml/pybnn.py:
````diff
@@ -0,0 +1,333 @@
+you need to install the following packages besides installing the hpolib.
+
+```
+pip3 install Theano==1.0.5
````
Can't you do `pip install.[pybnn]?`

KEggensperger commented on hpolib/benchmarks/ml/pybnn.py:
```diff
@@ -0,0 +1,333 @@
+    @staticmethod
+    def get_meta_information():
+        return {'name': 'BNN Benchmark',
+                'references': []
```
The reference here should be the BOHB paper as well, right?

PhMueller committed:
ADD reference to BOHB

PhMueller committed:
FIX in tests

the tests take more time on travis. Check if the cost of a benchmark is simply > 1. (it should not be 0 or None)

PhMueller committed:
FIX in tests

- add availability test - check if we can find the container in the library.
- add meta data to pybnn and cartpole.py
- remove currently the nasbench201 tests

PhMueller committed:
FIX Pytest8 error
